### ISSUE TITLE
Krea 2 text encoder silently truncates captions longer than 512 tokens

### ISSUE THREAD

**Issue by Xuness** [kohya-ss/musubi-tuner#1002]
## Tested revision and environment

- musubi-tuner: `v0.3.4`, commit [`30c658c`](https://github.com/kohya-ss/musubi-tuner/commit/30c658c4f4b0bf05038b3346eff9670259b10fc7)
- OS: Windows 11 Pro, build 26200
- Python: 3.11.9
- PyTorch: 2.12.1+cu130
- CUDA: 13.0 (driver 610.62)
- GPU: NVIDIA GeForce RTX 5090 D, 32 GB

## Problem

The Krea 2 Qwen3-VL conditioner currently defaults to a text length of 512 tokens. During text-encoder cache generation, longer captions are silently truncated, so the DiT never sees the complete caption even though Qwen3-VL supports substantially longer contexts.

The current fixed-length padding also makes simply raising the limit impractical: padding every prompt to a large fixed length would waste text-encoder memory.

## Steps to reproduce

1. On the tested revision, prepare an image caption whose Krea 2 Qwen3-VL encoded prompt is longer than 512 tokens.
2. Generate the text-encoder cache normally, for example:

   ```shell
   python src/musubi_tuner/krea2_cache_text_encoder_outputs.py \
     --dataset_config <dataset.toml> \
     --text_encoder <qwen3-vl-4b-instruct.safetensors>
   ```

3. Compare the valid sequence stored in the resulting `*_kr2_te.safetensors` cache with the untruncated tokenizer result.

### Actual behavior

`TextEncoderConfig.max_length` defaults to 512 and the conditioner calls the tokenizer with `truncation=True` and `padding="max_length"`. Tokens beyond that limit are absent from the cached conditioning without an error or warning.

### Expected behavior

It should be possible to cache the complete caption up to a configurable maximum, without forcing every short prompt in the batch to be padded to that maximum.

## Real-data evidence

In one real dataset validation:

- 218 captions were scanned.
- 96 captions exceeded the current 512-token limit.
- The longest valid encoded length was approximately 1,319 tokens.
- After applying the local fix, a generated varlen cache contained `varlen_krea2_vl_embed_bfloat16` with shape `(1319, 12, 2560)`, matching the tokenizer result.

This confirms that the previous limit discarded meaningful caption content and that the existing varlen cache format can store the longer result without fixed-length padding on disk.

## Proposed solution

- Make the Krea 2 text-encoder maximum length configurable through `--krea2_text_encoder_max_length`.
- Raise the supported default maximum to 16,384 tokens.
- Pad each encoder batch only to its longest prompt instead of padding every prompt to the configured maximum.
- Pass the configured value through the Krea 2 text-encoder loading entry points used for cache generation, training-time sample prompts, and standalone generation.
- Keep saving only valid tokens in the existing varlen cache format.

## Compatibility and memory impact

Short captions keep the same effective embeddings and varlen cache layout. Captions longer than 512 tokens will produce longer contexts, which can materially increase text-encoder and DiT activation memory. Users training with long captions may need gradient checkpointing or a smaller batch size.

The main compatibility question is the default:

- Should Krea 2 default to 16,384 so captions are not silently truncated?
- Or should the default remain 512 for memory compatibility, with longer contexts enabled explicitly through the new option?

I have a local implementation and tests ready and am happy to adjust the default based on maintainer preference.
